lib/Minz/Log.php

```diff
@@ -70,12 +71,44 @@ public static function record ($information, $level, $file_name = null) {
 			     . ' [' . $level_label . ']'
 			     . ' --- ' . $information . "\n";
 
+			self::ensureMaxLogSize($file_name);
+
 			if (file_put_contents($file_name, $log, FILE_APPEND | LOCK_EX) === false) {
 				throw new Minz_PermissionDeniedException($file_name, Minz_Exception::ERROR);
 			}
 		}
 	}
 
+	/**
+	 * Make sure we do not waste a huge amount of disk space with old log messages.
+	 *
+	 * This method can be called multiple times for one script execution, but its result will not change unless
+	 * you call clearstatcache() in between. We won't due do that for performance reasons.
+	 *
+	 * @param $file_name
+	 * @throws Minz_PermissionDeniedException
+	 */
+	protected static function ensureMaxLogSize($file_name) {
+		$maxSize = defined('MAX_LOG_SIZE') ? MAX_LOG_SIZE : 1048576;
+		if ($maxSize > 0 && @filesize($file_name) > $maxSize) {
```

Comment:
To be honest, I would still solve that in a different way, all these low level calls make me nervous... if you really want to keep the old logs, why not just rename the "current.log" to "backup.log". The task would be done in one call and less magic.

Reply:
Then the logic would be more complicated to show the logs in the UI, needing to read more than one log. Similarly, the logic for logrotate would also be more complicated. Furthermore, it would only push the problem further, by not deleting old logs if we keep them under a new name.